cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -0,0 +1,850 @@
+-----------------------------------------------------------------------------------------
+
+-- | Please note that there's a difference between Cardano tx and fiscal tx:
+--   1. Cardano tx is a transaction from Cardano blockchain's point of view.
+--   2. Fiscal tx is a transaction from recipient's point of view.
+--   So if one Cardano tx contains 10 outputs (with addresses of 10 recipients),
+--   we have 1 Cardano tx and 10 fiscal txs.
+runBenchmark
+  :: RunDemo (ByronBlockOrEBB ByronConfig)
+  => Tracer IO (TraceBenchTxSubmit (Byron.GenTxId (ByronBlockOrEBB ByronConfig)))
+  -> Tracer IO SendRecvConnect
+  -> Tracer IO (SendRecvTxSubmission (ByronBlockOrEBB ByronConfig))
+  -> Tracer IO String
+  -> CardanoConfiguration
+  -> NodeConfig ByronEBBExtNodeConfig
+  -> CC.Common.Address
+  -> Crypto.SigningKey
+  -> CC.Common.Address
+  -> TopologyInfo
+  -> NumberOfTxs
+  -> NumberOfOutputsPerTx
+  -> FeePerTx
+  -> TPSRate
+  -> IO ()
+runBenchmark tracer
+             connectTracer
+             submitTracer
+             lowLevelSubmitTracer
+             cc
+             pInfoConfig
+             _sourceAddress
+             sourceKey
+             recipientAddress
+             topologyInfo
+             numOfTxs
+             numOfOutsPerTx
+             txFee
+             tpsRate = do
+  traceWith tracer . TraceBenchTxSubDebug $ "******* Tx generator, phase 1: make enough available UTxO entries *******"
+  createMoreFundCoins lowLevelSubmitTracer
+                        cc
+                        pInfoConfig
+                        sourceKey
+                        txFee
+                        topologyInfo
+                        numOfTxs
+
+  traceWith tracer . TraceBenchTxSubDebug $ "******* Tx generator, phase 2: pay to recipients *******"
+  let benchmarkTracers :: BenchmarkTxSubmitTracers IO (ByronBlockOrEBB ByronConfig)
+      benchmarkTracers = BenchmarkTracers
+                           { trSendRecvConnect      = connectTracer
+                           , trSendRecvTxSubmission = submitTracer
+                           }
+
+  let localAddr :: Maybe Network.Socket.AddrInfo
+      localAddr = Nothing
+
+  let hints :: AddrInfo
+      hints = defaultHints
+        { addrFlags      = [AI_PASSIVE]
+        , addrFamily     = AF_INET6
+        , addrSocketType = Stream
+        , addrCanonName  = Nothing
+        }
+  -- Corresponds to a node 0 (used by default).
+  (remoteAddr:_) <- getAddrInfo (Just hints) (Just "::1") (Just "3000")
+
+  let updROEnv
+        :: ROEnv (Byron.GenTxId (ByronBlockOrEBB ByronConfig)) (GenTx (ByronBlockOrEBB ByronConfig))
+        -> ROEnv (Byron.GenTxId (ByronBlockOrEBB ByronConfig)) (GenTx (ByronBlockOrEBB ByronConfig))
+      updROEnv defaultROEnv =
+        ROEnv { targetBacklog     = targetBacklog defaultROEnv
+              , txNumServiceTime  = Nothing
+              , txSizeServiceTime = Nothing
+              }
+
+  -- Run generator.
+  txGenerator tracer
```

Comment:
Maybe a `data` type for `txGenerator` as well? `TxGeneratorParams` or somesuch..